backend/battle/battles/battle.py

```diff
@@ -29,14 +31,57 @@ def get_winner_for(team_creator, team_opponent):
 
 
 def validate_sum_pokemons(pokemons):
+    pokemons_saved = []
+    total_api_pokemon_point = 0
+    pokemons_not_saved = []
     total_points = 0
     for pokemon in pokemons:
-        pokemon_point = pokemon.attack + pokemon.defense + pokemon.hp
+        on_database = verify_pokemon_exists_on_database(pokemon)
+        if not on_database:
+            pokemons_not_saved.append(pokemon)
+            each_api_pokemon_point = sum_point_from_api(pokemon)
+            total_api_pokemon_point += each_api_pokemon_point
+        else:
+            pokemons_saved.append(on_database)
+    for pokemon in pokemons_saved:
+        pokemon_point = sum_points_pokemon_on_database(pokemon)
         total_points += pokemon_point
+    if total_api_pokemon_point:
+        total_points += total_api_pokemon_point
+    if total_points <= 600:
+        for pokemon in pokemons_not_saved:
+            create_pokemon_on_database(pokemon)
```

Comment:
This logic is a bit confusing. You're at the same time validating and saving?
Consider separating those things, like:
1. First save what is not saved (separate this on another function)
2. Then have a single logic to sum those points

Reply:
Well, I need to sum those points before saving the pokemon from API, because if the points sum more than 600, I cannot save the pokemon selected

Reply:
Now I understood your argument. After your advice, I think I fixed this misleading

Reply:
Thank you!

Reply:
@rsarai could you please review the new logic?
Code-wise is fine, but I'm not sure if the behavior needs to be like this.